execution_engine/src/storage/trie_store/operations/mod.rs

```diff
@@ -394,6 +479,238 @@ where
+                                pointer_block[idx as usize] = Some(sibling_pointer);
+                                let trie_node = Trie::Node { pointer_block };
+                                let trie_key = Blake2bHash::new(&trie_node.to_bytes()?);
+                                new_elements.push((trie_key, trie_node))
+                            }
+                        }
+                    }
+                    // The single sibling is a node or an extension, and a grandparent exists.
+                    // Therefore the parent is not the root
+                    (Pointer::NodePointer(sibling_trie_key), Some((idx, grandparent))) => {
+                        // Push the grandparent back onto the parents so it may be processed later.
+                        parents.push((idx, grandparent));
+                        // Elsewhere we maintain the invariant that all trie keys have corresponding
+                        // trie values.
+                        let sibling_trie = store
+                            .get(txn, &sibling_trie_key)?
+                            .expect("should have sibling");
+                        match sibling_trie {
+                            Trie::Leaf { .. } => {
+                                panic!("Node pointer should not point to leaf")
+                            }
+                            // The single sibling is a node, and the grandparent does not exist.
+                            // Therefore the parent is the root.  We output an extension to
+                            // replace the parent, with a single byte corresponding to the sibling
+                            // index.  In the next loop iteration, we will handle the case where
+                            // this extension might need to be combined with a grandparent
+                            // extension.
+                            Trie::Node { .. } => {
+                                let new_extension: Trie<K, V> = Trie::Extension {
+                                    affix: vec![sibling_idx].into(),
+                                    pointer: sibling_pointer,
+                                };
+                                let trie_key = Blake2bHash::new(&new_extension.to_bytes()?);
+                                new_elements.push((trie_key, new_extension))
+                            }
+                            // The single sibling is a node, and there exists a grandparent.
+                            // Therefore the parent is not the root.  We output an extension to
+                            // replace the parent, with a single byte corresponding to the sibling
+                            // index.  In the next loop iteration, we will handle the case where
+                            // this extension might need to be combined with a grandparent
+                            // extension.
+                            Trie::Node { .. } => {
```

Comment:
Isn't this identical to the previous match arm? (Was the distinction about the grandparents removed at some point?)

Reply:
Yes, they are identical.   Fixed in https://github.com/CasperLabs/casper-node/pull/1155/commits/e0718bb1afd75d046aa458c8647f39714531e3b1